check-mysql/lib/check-mysql.go

```diff
@@ -37,6 +40,44 @@ var commands = map[string](func([]string) *checkers.Checker){
 	"readonly":    checkReadOnly,
 }
 
+func readCnf(m mysqlSetting) (mysqlSetting, error) {
+	cfg, err := ini.LoadSources(
+		ini.LoadOptions{
+			AllowBooleanKeys: true,
+			Insensitive:      true,
+			Loose:            true,
+		},
+		m.Cnf,
+	)
+	if err != nil {
+		return m, err
+	}
+	if cfg != nil {
+		for _, s := range cfg.Sections() {
+			if s.Name() == m.Profile {
+				if s.Key("host").Value() != "" {
+					m.Host = s.Key("host").Value()
+				}
+				if s.Key("port").Value() != "" {
+					m.Port = s.Key("port").Value()
+				}
+				if s.Key("socket").Value() != "" {
+					m.Socket = s.Key("socket").Value()
+				}
+				if s.Key("user").Value() != "" {
+					m.User = s.Key("user").Value()
+				}
+				if s.Key("password").Value() != "" {
+					m.Pass = s.Key("password").Value()
+				}
```

Comment:
If the configuration file and the argument options are listed together, I want the argument options to take precedence.

Reply:
Makes a lot of sense!

Reply:
We will do our best to accept the changes.
If the work seems to be difficult, it would be helpful if you could write a test of the behavior added this time.

Reply:
@GermanG 
Happy New Year!
What happened to this pull request?

Reply:
Thanks, same to you!
No news so far. I didn't like my first approach to this, I'm reworking it.

Reply:
> I'm reworking it.

sure.
I think it would be better if the values are applied in the order of default, my.cnf reference, and arguments.